metis-technical-metadata-generation/src/main/java/eu/europeana/metis/technical/metadata/generation/utilities/MediaExtractorForFile.java

```diff
@@ -0,0 +1,174 @@
+
+  MediaExtractorForFile(File datasetFile, MongoDao mongoDao, MediaExtractor mediaExtractor,
+      boolean startFromBeginningOfFiles, boolean retryFailedResources) {
+    this.datasetFile = datasetFile;
+    this.mongoDao = mongoDao;
+    this.mediaExtractor = mediaExtractor;
+    this.startFromBeginningOfFiles = startFromBeginningOfFiles;
+    this.retryFailedResources = retryFailedResources;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    return parseMediaForFile(datasetFile);
+  }
+
+  private Void parseMediaForFile(File datasetFile) throws IOException {
+    LOGGER.info(EXECUTION_LOGS_MARKER, "Parsing: {}", datasetFile);
+    final FileStatus fileStatus = getFileStatus(datasetFile.getName(), startFromBeginningOfFiles,
+        retryFailedResources);
+
+    int lineIndex = fileStatus.getLineReached();
+    try (Scanner scanner = new Scanner(datasetFile, "UTF-8")) {
+      //Bypass lines until the reached one, from a previous execution
+      if (moveScannerToLine(scanner, fileStatus)) {
+        while (scanner.hasNextLine()) {
+          String resourceUrl = scanner.nextLine();
+          LOGGER.info(EXECUTION_LOGS_MARKER, "Processing resource: {}", resourceUrl);
+          //Should this resource be processed according to status and configuration parameters
+          if (eligibleForProcessing(resourceUrl)) {
+            final ResourceExtractionResult resourceExtractionResult;
+            try {
+              resourceExtractionResult = performMediaExtraction(resourceUrl);
+              mongoDao.storeMediaResultInDb(resourceExtractionResult);
+              clearThumbnails(resourceExtractionResult);
+            } catch (MediaExtractionException e) {
+              LOGGER.warn(EXECUTION_LOGS_MARKER, "Media extraction failed for resourceUrl {}",
+                  resourceUrl);
+              mongoDao.storeFailedMediaInDb(resourceUrl);
+            }
+          } else {
+            LOGGER.info(EXECUTION_LOGS_MARKER, "ResourceUrl already exists in db: {}", resourceUrl);
+          }
+          lineIndex++;
+          fileStatus.setLineReached(lineIndex);
+          mongoDao.storeFileStatusToDb(fileStatus);
+
+          if (lineIndex % 100 == 0) {
+            LOGGER.info(EXECUTION_LOGS_MARKER, "Processing file: {}, reached line {}",
+                datasetFile.getName(), lineIndex);
+          }
+        }
+        fileStatus.setEndOfFileReached(true);
+        mongoDao.storeFileStatusToDb(fileStatus);
+      }
+    }
+    LOGGER.info(EXECUTION_LOGS_MARKER, "Finished: {}", datasetFile);
+    return null;
+  }
+
+  private ResourceExtractionResult performMediaExtraction(String resourceUrl)
+      throws MediaExtractionException {
+    //Use all url types to get metadata and thumbnails for all. Later we decide what to use.
+    RdfResourceEntry resourceEntry = new RdfResourceEntry(resourceUrl,
+        Arrays.asList(UrlType.OBJECT, UrlType.HAS_VIEW, UrlType.IS_SHOWN_BY, UrlType.IS_SHOWN_AT));
```

Comment:
Use Arrays.asList(UrlType.values()) or, even better, UrlType.URL_TYPES_FOR_MEDIA_EXTRACTION.
This would protect the code in the (remote) possibility that a new UrlType is added.

Reply:
Fixed